lib/semian/grpc.rb

```diff
@@ -0,0 +1,80 @@
+require 'semian/adapter'
+require 'grpc'
+
+module GRPC
+  GRPC::BadStatus.include(::Semian::AdapterError)
+
+  class SemianError < GRPC::BadStatus
```

Comment:
This seems problematic for Panama because it [assumes a detail is included for BadStatus](https://github.com/Shopify/panama/blob/master/ruby/lib/shopify_panama/errors.rb). So if we choose this error, we need `#details` to be implemented.

Reply:
Ok, I'll check back when you have :)

Reply:
Isn't there some kind of `BaseConnectionError`, it's usually the ones for the best semantic for semian.

Reply:
From looking at https://github.com/grpc/grpc/blob/master/src/ruby/lib/grpc/errors.rb there is nothing similar to `BaseConnectionError` 😞 . The main reason I picked `BadStatus` is that it seemed to be the base class of all the errors, I am not sure which error would be the best based on the error list. From the list, are you seeing a better one?

Reply:
I'm not totally sure of the semantic of each, but `Unavailable` seems like the best candidate.

For context we want Semian exceptions to be handled by the caller like they would handle connection issues. 

So if they have code like this:

```ruby
begin
  grpc.call
rescue GRPC::NotFound
  return nil
rescue GRPC::Unavailable
  # we want Semian exceptions to trigger that handler.
end
```

This way people don't have to explicitly handle semian, it's just a special case of a regular network issue.

Now it's best to check whatever error the library send if it can't connect, but I suppose it's `Unavailable`

Reply:
I've updated it to `Unavailable` If the client can't connect to the server, it will, in fact, return an `Unavailable` error. 👍